@here/harp-examples/src/datasource_object-picking.ts

```diff
@@ -0,0 +1,144 @@
+                padding: 9px 12px;
+                background: #37afaa;
+                display: inline-block;
+                visibility: hidden;
+            }
+            #mapCanvas {
+              top: 0;
+            }
+        </style>
+
+        <pre id="mouse-picked-result"></pre>
+    `;
+
+    initializeMapView("mapCanvas").catch(err => {
+        throw err;
+    });
+
+    // snippet:datasource_object_picking_2.ts
+    const element = document.getElementById("mouse-picked-result") as HTMLPreElement;
+    let currentUserData: PickResult | undefined;
+
+    function handlePick(mapViewUsed: MapView, x: number, y: number) {
+        // get an array of intersection results from MapView
+
+        let usableIntersections = mapViewUsed
+            .intersectMapObjects(x, y)
+            .filter(item => item.userData !== undefined);
+        if (usableIntersections.length > 1) {
+            usableIntersections = usableIntersections.filter(
+                item => item.userData !== currentUserData
+            );
+        }
+
+        if (usableIntersections.length === 0) {
+            // Hide helper box
+            element.style.visibility = "hidden";
+            return;
+        }
+
+        // Get userData from the first result;
+        currentUserData = usableIntersections[0].userData;
+
+        // Show helper box
+        element.style.visibility = "visible";
+
+        // Display userData inside of helper box
+        element.innerText = JSON.stringify(currentUserData, undefined, 2);
+    }
+    // end:datasource_object_picking_2.ts
+
+    /**
+     * Creates a new MapView for the HTMLCanvasElement of the given id.
+     */
+    async function initializeMapView(id: string) {
+        const canvas = document.getElementById(id) as HTMLCanvasElement;
+        const mapView = new MapView({
+            canvas,
+            theme: "resources/berlin_tilezen_base.json",
+            enableRoadPicking: true
+        });
+
+        CopyrightElementHandler.install("copyrightNotice")
+            .attach(mapView)
+            .setDefaults([
+                {
+                    id: "openstreetmap.org",
+                    label: "OpenStreetMap contributors",
+                    link: "https://www.openstreetmap.org/copyright"
+                }
+            ]);
+
+        const controls = new MapControls(mapView);
+
+        // Add an UI.
+        const ui = new MapControlsUI(controls);
+        canvas.parentElement!.appendChild(ui.domElement);
+
+        window.addEventListener("resize", () => {
+            mapView.resize(window.innerWidth, window.innerHeight);
+        });
+
+        // snippet:datasource_object_picking_1.ts
+        canvas.addEventListener("mousedown", event => {
+            handlePick(mapView, event.pageX, event.pageY);
+        });
+        // end:datasource_object_picking_1.ts
+
+        const omvDataSource = new OmvDataSource({
+            baseUrl: "https://xyz.api.here.com/tiles/osmbase/512/all",
```

Comment:
Why not HERE VectorTiles (i.e. `https://xyz.api.here.com/tiles/herebase.02`)?

Reply:
I only used a piece of code from other examples, but thanks for the hint using this data source found a bug HARP-7129.

Reply:
Done